domains/webhooks.go

```diff
@@ -0,0 +1,97 @@
+package domains
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/spf13/viper"
+)
+
+// Webhook represents webhook.
+type Webhook struct {
+	Url               string    `json:"url"`
+	TokenHeader       string    `json:"-"`
+	Token             string    `json:"-"`
+	CreatedAt         time.Time `json:"createdAt"`
+	LastEmitStatus    string    `json:"lastEmitStatus"`
+	LastEmitTimestamp time.Time `json:"lastEmitTimestamp"`
+	ErrorsCount       int       `json:"errorsCount"`
+	Active            bool      `json:"active"`
+}
+
+// WebhookMaxTries is the maximum number of times a webhook will be retried.
+const WebhookMaxTries = "webhook.maxTries"
+
+// WebhookTargetClient is the interface for the webhooks http calls.
+type WebhookTargetClient interface {
+	Call(headers map[string]string, method string, url string, body *BlockHeader) (*http.Response, error)
+}
+
+// Notify sends notification to webhook.
+func (w *Webhook) Notify(h *BlockHeader, client WebhookTargetClient) error {
+	// Prepare headers
+	headers := map[string]string{
+		w.TokenHeader:  w.Token,
+		"Content-Type": "application/json",
+	}
+
+	res, err := client.Call(headers, http.MethodPost, w.Url, h)
+
+	if err != nil {
+		// Update the webhook after failed notification.
+		w.updateWebhookAfterNotification(0, "", err)
+		return err
+	}
+
+	defer res.Body.Close() // nolint: all
+
+	// Read the response.
+	body, _ := io.ReadAll(res.Body)
```

Comment:
you should assign err here 
```suggestion
	body, err := io.ReadAll(res.Body)
```